tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2241,6 +2241,16 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id, mdMet
     pNewInstr->m_opcode = CEE_RET;
     rewriter_void.InsertBefore(pFirstInstr, pNewInstr);
 
+    if (IsDumpILRewriteEnabled())
+    {
+        mdToken token = 0;
+        TypeInfo typeInfo{};
+        WSTRING methodName = WStr("__DDVoidMethodCall__");
+        FunctionInfo caller(token, methodName, typeInfo, MethodSignature(), FunctionMethodSignature());
+        Logger::Info(
+            GetILCodes("*** GenerateVoidILStartupMethod(): Modified Code: ", &rewriter_void, caller, metadata_import));
+    }
```

Comment:
Yes, I wanted to use our `GetILCodes` function to emit the IL for the startup hook to improve our diagnostics, but I needed to pass in an instance of the `metadata_import` interface to resolve method names from the metadata tokens in the IL. So I passed that down to this call and refactored `GetILCodes` to accept the `metadata_import` interface instead of our full `module_metadata` structure, which is not available at this specific callsite.